CONTRIBUTING.md

```diff
@@ -69,13 +63,152 @@ README for more details.
+analytics.track_event('Some Event Name', result.to_h)
+
+if result.success?
+  handle_success
+else
+  handle_failure
+end
+```
+
+* Only make one call to `analytics.track_event` after submitting the form, as
+opposed to one call when handling success and another when handling failure. The
+Form Object, when used properly, will return a FormResponse object that already
+tells us whether the action was successful or not.
+
+### Importance of the controller design for analytics
+
+This design pattern was the result of many iterations, and agreed upon by all
+team members in early 2017. It keeps controllers clean and predictable. Having a
+controller interact with a Form Object or some other specialized class is not a
+new concept. Rails developers have been using them since at least 2012. What
+might seem new is the `FormResponse` object. **The most important reason
+controllers expect an object that responds to `success?` and `to_h` is to define
+an analytics API, or a contract, if you will, between the analytics logs and the
+folks who query them.**
+
+For example, if someone wants to look up all events that have failed, they would
+run this query in Kibana: `properties.event_properties.success:false`. Now let's
+say a developer introduces a new controller that doesn't adhere to our established
+convention, and captures analytics in their own way, without adding `success`
+and `errors` keys, which are expected to be included in all analytics events.
+This means that any failures for this controller won't show up when running the
+query above, and the person running the query might not realize data is missing.
+
+Deviating from the convention also causes confusion. The next developer to join
+the team will not be sure which pattern to use, and might end up picking the
+wrong pattern. As Sandi Metz says:
+
+> For better or for worse, the patterns you establish today will be replicated
+forever. When the code lies you must be alert to programmers believing and then
+propagating that lie.
+
+### Secure controllers
+Rails by default is currently vulnerable to [cache poisoning attacks]
+through modification of the `X-Forwarded-For` and `Host` headers. In
+order to protect against the latter, there are two pieces that must be
+in place. The first one is already taken care of by defining
+`default_url_options` in `ApplicationController` with a `host` value
+that we control.
+
+The other one is up to you when adding or modifying redirects:
+
+- Always use `_url` helpers (as opposed to `_path`) when calling
+`redirect_to` in a controller.
+
 ### Additional notes on pull requests and code reviews
 
+Please follow our [Code Review][review] guidelines.
+[Glen Sanford's thoughts on code reviews][thoughts] are also well worth
+reading.
+
+[review]: https://github.com/18F/development-guide/tree/master/code_review
+[thoughts]: http://glen.nu/ramblings/oncodereview.php
+
 - Prioritize code reviews above your other work
```

Comment:
Prioritize code reviews of issues in the sprint above other work

Reply:
Yes, great clarification